src/redismodule.h

```diff
@@ -1,1695 +1,719 @@
-    REDISMODULE_GET_API(CommandFilterGetClientId);
-    REDISMODULE_GET_API(Fork);
-    REDISMODULE_GET_API(SendChildHeartbeat);
-    REDISMODULE_GET_API(ExitFromChild);
-    REDISMODULE_GET_API(KillForkChild);
-    REDISMODULE_GET_API(GetUsedMemoryRatio);
-    REDISMODULE_GET_API(MallocSize);
-    REDISMODULE_GET_API(MallocUsableSize);
-    REDISMODULE_GET_API(MallocSizeString);
-    REDISMODULE_GET_API(MallocSizeDict);
-    REDISMODULE_GET_API(CreateModuleUser);
-    REDISMODULE_GET_API(FreeModuleUser);
-    REDISMODULE_GET_API(SetContextUser);
-    REDISMODULE_GET_API(SetModuleUserACL);
-    REDISMODULE_GET_API(SetModuleUserACLString);
-    REDISMODULE_GET_API(GetModuleUserACLString);
-    REDISMODULE_GET_API(GetCurrentUserName);
-    REDISMODULE_GET_API(GetModuleUserFromUserName);
-    REDISMODULE_GET_API(ACLCheckCommandPermissions);
-    REDISMODULE_GET_API(ACLCheckKeyPermissions);
-    REDISMODULE_GET_API(ACLCheckChannelPermissions);
-    REDISMODULE_GET_API(ACLAddLogEntry);
-    REDISMODULE_GET_API(ACLAddLogEntryByUserName);
-    REDISMODULE_GET_API(DeauthenticateAndCloseClient);
-    REDISMODULE_GET_API(AuthenticateClientWithACLUser);
-    REDISMODULE_GET_API(AuthenticateClientWithUser);
-    REDISMODULE_GET_API(RedactClientCommandArgument);
-    REDISMODULE_GET_API(GetClientCertificate);
-    REDISMODULE_GET_API(GetCommandKeys);
-    REDISMODULE_GET_API(GetCommandKeysWithFlags);
-    REDISMODULE_GET_API(GetCurrentCommandName);
-    REDISMODULE_GET_API(RegisterDefragFunc);
-    REDISMODULE_GET_API(DefragAlloc);
-    REDISMODULE_GET_API(DefragRedisModuleString);
-    REDISMODULE_GET_API(DefragShouldStop);
-    REDISMODULE_GET_API(DefragCursorSet);
-    REDISMODULE_GET_API(DefragCursorGet);
-    REDISMODULE_GET_API(GetKeyNameFromDefragCtx);
-    REDISMODULE_GET_API(GetDbIdFromDefragCtx);
-    REDISMODULE_GET_API(EventLoopAdd);
-    REDISMODULE_GET_API(EventLoopDel);
-    REDISMODULE_GET_API(EventLoopAddOneShot);
-    REDISMODULE_GET_API(RegisterBoolConfig);
-    REDISMODULE_GET_API(RegisterNumericConfig);
-    REDISMODULE_GET_API(RegisterStringConfig);
-    REDISMODULE_GET_API(RegisterEnumConfig);
-    REDISMODULE_GET_API(LoadConfigs);
-    REDISMODULE_GET_API(RdbStreamCreateFromFile);
-    REDISMODULE_GET_API(RdbStreamFree);
-    REDISMODULE_GET_API(RdbLoad);
-    REDISMODULE_GET_API(RdbSave);
-
-    if (RedisModule_IsModuleNameBusy && RedisModule_IsModuleNameBusy(name)) return REDISMODULE_ERR;
-    RedisModule_SetModuleAttribs(ctx,name,ver,apiver);
-    return REDISMODULE_OK;
-}
-
-#define RedisModule_Assert(_e) ((_e)?(void)0 : (RedisModule__Assert(#_e,__FILE__,__LINE__),exit(1)))
-
-#define RMAPI_FUNC_SUPPORTED(func) (func != NULL)
+ * The above directive will pull in the VALKEYMODULE_* definitions while
+ * ensuring that any reference to REDISMODULE_* in the module's source
+ * will resolve to the appropriate VALKEYMODULE_* counterpart.
+ */
 
-#endif /* REDISMODULE_CORE */
-#endif /* REDISMODULE_H */
+#include "valkeymodule.h"
```

Comment:
Just a note about when we write compatibility. This will not always work, if someone was just copying

Reply:
Not a big deal IMO. Users can copy it from redis 7.2.4 instead.

Reply:
I am also less concerned about this. the (build) errors are self-explanatory IMO. I also left sufficient comments in redismodule.h to explain its purpose. Lastly, I would expect module developers to use the redismodule.h copy from the redis tree to build redis modules.